### PR TITLE
GaaS UI Dependency Upgrades

### PR DESCRIPTION
**Pull request by K-itKat20**
Upgraded dependencies in GaaS UI, Including testing it on node v16 and v17 (#162 #193)

### CODE REVIEW

File: gaas-ui/src/setupProxy.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const { createProxyMiddleware } = require("http-proxy-middleware");
+const { createProxyMiddleware } = require('http-proxy-middleware');
```
Comment (t92549) [inline gaas-ui/src/setupProxy.js:1]: Why has this changed to single quotes?

File: gaas-ui/test/setupTests.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import Enzyme from "enzyme";
-import Adapter from "enzyme-adapter-react-16";
+import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
```
Comment (t92549) [inline gaas-ui/test/setupTests.ts:2]: Is this update unofficial?

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+coverage:
+  status:
+    project:
+      default:
+        informational: true
+    patch:
+      default:
+        informational: true
```
Comment (t92549) [inline codecov.yml:8]: Same comment as in https://github.com/gchq/gaffer-experimental/pull/209:
Missing newline

File: .github/workflows/CodeAnalysisAndCI.yml [hunk 4]
```diff
@@ -36,14 +36,19 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           flags: unittests # optional
           name: jacoco-report-uploader # optional
-          fail_ci_if_error: true # optional (default = false)
```
Comment (t92549) [inline .github/workflows/CodeAnalysisAndCI.yml:39]: Same comment as in https://github.com/gchq/gaffer-experimental/pull/209:
Does this need to go?

**Review by t92549 — CHANGES_REQUESTED** [review]
Thanks for the PR

File: codecov.yml (hunk 3 above)
Comment (rocky341) [reply codecov.yml:8]: Added

File: .github/workflows/CodeAnalysisAndCI.yml (hunk 4 above)
Comment (rocky341) [reply .github/workflows/CodeAnalysisAndCI.yml:39]: Have responded in #209 let me know your thoughts please

File: gaas-ui/test/setupTests.ts (hunk 2 above)
Comment (rocky341) [reply gaas-ui/test/setupTests.ts:2]: It looks to be an unofficial adaptor

Comment (rocky341) [reply gaas-ui/test/setupTests.ts:2]: Have just had a quick google and found this:
https://stackoverflow.com/questions/64658031/which-enzyme-adapter-works-with-react-17

Comment (t92549) [reply gaas-ui/test/setupTests.ts:2]: Is this fine to use then? Wondering on React 16 vs 17?

**Comment by t92549** [comment]
Sorry this now needs a conflict resolving @rocky341

File: gaas-ui/test/setupTests.ts (hunk 2 above)
Comment (rocky341) [reply gaas-ui/test/setupTests.ts:2]: Your call on that one. The aim has been to upgrade the versions of different frameworks in order to minimise possible vulnerabilities. It seems as though this adaptor is the one which gets used when using react 17. Think 17 doesn't have a huge deal of changes over 16. The main one seems to be that in the future upgrading from 17 up is easier than in previous iterations.

Comment (rocky341) [reply gaas-ui/test/setupTests.ts:2]: My opinion would be that its okay and it doesn't seem like enzyme is going to release an official adaptor anytime soon for react 17.

**Review by t92549 — APPROVED** [review]

**Pull request merged**